src/scaffold/entry_type/utils.rs

```diff
@@ -50,6 +41,7 @@ fn inner_choose_referenceable(
     all_options.push("Agent".to_string());
 
     if optional {
+        all_options.push("AnyLinkableHash".to_string());
```

Comment:
I would expect the `AnyLinkableHash` to also be available in the non-optional scenario, or is there a reason for it not to be available?

Reply:
The reason that it's not available is to only allow setting an `ExternalHash` (prev `AnyLinkableHash`) as a target rather than base. This just for semantical reasons really, what do you think?

Reply:
I think there are also cases where you might want to have an external hash as a base...